### PR TITLE
Dependencies update

### PR DESCRIPTION
**Pull request by lindsay-stevens**
#### What has been done to verify that this works as intended?

Ran tests, linter, and checked compiled docs.

#### Why is this the best possible solution? Were any other approaches considered?

That time of year again for moving up python versions and doing some housekeeping. Due to the ruff rule changes it'd be easier to review per-commit.

- Move python support window from 3.10 to 3.13 to 3.11 to 3.14. Python 3.10 is fully EOL on 2026-10-31, and 3.14 would be stable now having been first released 2025-10-07. 
- Update / apply ruff rules. Most of the changes are docs improvements.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

As noted above.

#### Do we need any specific form for testing your changes? If so, please attach one.

No

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.

No

#### Before submitting this PR, please make sure you have:
- [x] included test cases for core behavior and edge cases in `tests`
- [x] run `python -m unittest` and verified all tests pass
- [x] run `ruff format pyodk tests` and `ruff check pyodk tests` to lint code
- [x] verified that any code or assets from external sources are properly credited in comments

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ This library aims to make common data analysis and workflow automation tasks as
 
 ## Install
 
-The currently supported Python versions for `pyodk` are 3.10 to 3.13 (the primary development version is 3.12). If this is different from the version you use for other projects, consider using [`pyenv`](https://github.com/pyenv/pyenv) to manage multiple versions of Python.
+The currently supported Python versions for `pyodk` are 3.11 to 3.14 (the primary development version is 3.13). If this is different from the version you use for other projects, consider using [`pyenv`](https://github.com/pyenv/pyenv) to manage multiple versions of Python.
 
 The currently supported Central version is v2025.1.4. Newer or older Central versions will likely work too, but convenience (non-HTTP) methods assume this version. If you see a 404 error or another server error, please verify the version of your Central server.
```
Comment (lognaturel) [inline README.md]: Could we also update this Central version? I think the intent is to keep this up to date as Central releases happen.

Comment (lognaturel) [reply README.md]: You're way ahead of me.

**Review by lognaturel — APPROVED** [review]
Thanks! Going to let you take a last look and merge since it's been sitting for a bit (sorry!).

Ah, and I did release v1.3.0 before this so there's a conflict.

**Comment by lindsay-stevens** [comment]
Rebased on master to resolve merge conflict.
